File: ios/xReader/Models/Book.swift
import Foundation

struct BookResponse: Codable, Identifiable {
    let id: Int
    let title: String
    let author: String?
    let format: String
    let file_path: String
    let cover_path: String?
    let chapter_count: Int
    let status: String
    let publish_year: Int?
    let created_at: String
    let updated_at: String
}

struct BookList: Codable {
    let items: [BookResponse]
    let total: Int
}

File: android/app/src/main/java/com/xreader/app/data/model/Book.kt
package com.xreader.app.data.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class BookResponse(
    val id: Int,
    val title: String,
    val author: String? = null,
    val format: String,
    @SerialName("file_path") val filePath: String,
    @SerialName("cover_path") val coverPath: String? = null,
    @SerialName("chapter_count") val chapterCount: Int,
    val status: String,
    @SerialName("publish_year") val publishYear: Int? = null,
    @SerialName("created_at") val createdAt: String,
    @SerialName("updated_at") val updatedAt: String
)

@Serializable
data class BookListResponse(
    val items: List<BookResponse>,
    val total: Int
)

@Serializable
data class BookUpdate(
    val title: String? = null,
    val author: String? = null
)

@Serializable
data class ReparseResponse(
    val message: String,
    @SerialName("chapter_count") val chapterCount: Int
)
